ecs-cli/integ/cmd_up/up_test.go

```diff
@@ -0,0 +1,149 @@
+// +build integ
+
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package cmd_up tests the "ecs-cli up" command with various configurations.
+package cmd_up
+
+import (
+	"fmt"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/integ"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	ecsCLIStackNamePrefix = "amazon-ecs-cli-setup-"
+)
+
+// maxNumberOfRetries * sleepDurationInBetweenRetriesInS represents how long we are willing
+// to wait before we fail a test
+const (
+	maxNumberOfRetries               = 10
```

Comment:
Good question! This number represents the amount of time that we're willing to validate for the container instance to become `ACTIVE` in the cluster. From my past runs, it looks like the instance comes up after ~210s = 3.5mins. Since 10*30 seconds = 5 mins >> 3.5 mins, it seemed like a safe number.

However, it looks like this is not clear and on top of it there is another const below `testTimeoutInMinutes` that is different than this cluster timeout.
I'll refactor the constants as follows:
```go
const (
  // How long we're willing to wait for the container instances to become ACTIVE in the cluster
  timeoutForWaitingOnActiveContainersInS = 300 

  // How long we expected the integration test to take 
  expectedTestDurationInMins = 15

  // How long we sleep in between retrying requests that fail
  sleepDurationInBetweenRetriesInS = 30
)
```